### PR TITLE
Consolidate RestResponse

### PR DESCRIPTION
**Pull request by RikkiGibson**
- Now only one RestResponse type which allows any combination of body/headers, only headers, only body, or neither to be specified. Raw headers are always present.

- Minor RestProxy refactor to enforce a notion of "unwrap the Single if it's there, unwrap the RestResponse if it's there, and now handle the body".

### CODE REVIEW

File: client-runtime/src/main/java/com/microsoft/rest/RestProxy.java [hunk 1]
```diff
@@ -353,7 +353,34 @@ public HttpResponse call(String responseBody) {
         return asyncResult;
     }
 
-    private Single<?> toProxyReturnValueAsync(final HttpResponse response, final SwaggerMethodParser methodParser, final Type entityType) {
+    private Single<?> handleRestResponseReturnTypeAsync(final HttpResponse response, final SwaggerMethodParser methodParser, final Type entityType) {
```
Comment (daschult) [inline client-runtime/src/main/java/com/microsoft/rest/RestProxy.java]: methodParser doesn't need to be `final` anymore.

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/RestProxy.java]: Neither does response.

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/RestProxy.java]: And neither does entityType.

Comment (RikkiGibson) [reply client-runtime/src/main/java/com/microsoft/rest/RestProxy.java]: Wouldn't it be nice if final were the default?

**Pull request merged**